vscode-trace-extension/src/utils/traceExtensionLogger.ts

```diff
@@ -0,0 +1,91 @@
+/***************************************************************************************
```

Comment:
Please rename `traceExtensionLogger.ts` to `trace-extension-logger.ts` to be consistent with other source code files

Most of the methods are not used. Could you please add some description to the message to know what each method does?

Questions:
- Does vscode make sure that panel is not overflowing?
- Does the panel need to be hidden or disposed at some point?

Reply:
I have added descriptions for the methods in the Logger.

Does vscode make sure that panel is not overflowing?
-> yes. There is also a method to clear the log channel both from the API and through the UI

 Does the panel need to be hidden or disposed at some point?
-> I have it disposing on extension deactivation now. Reason being my thinking is that this channel can be used by all the modules of the extension, not just the file handler. As for showing/hiding it, the current API only allows hiding the entire outputs panel, and not the individual channels. Hiding individual channels and replacing it with the previously shown channel would be more ideal but its the current limitation of the vscode api. For now, we can just show the output log channel whenever we log something to it and not worry about hiding it. Other extensions that use output channels would do the same.

->The alternatives are always keeping it hidden, and the user can decide when they want to see the trace extension channel by selecting it from the dropdown menu in the outputs panel. The other option is actually using the hide channel api call, which will hide the entire outputs panel (not my preference).

WDYT?

Reply:
I think opening the log window everytime we log something might be a bit disruptive, because I would have to close it everytime manually when I want to get more vertical space. I understand that it would be less discoverable for end-users, but I think it's better. I don't think other extension open it automatically.

Reply:
yeah i agree with that. We can just log to it, and if user wants to inspect it, they can open the channel.